pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -194,13 +194,13 @@ func convertServices(cfg config.Config) []*model.Service {
 	}
 
 	out = append(out, buildServices(hostAddresses, cfg.Namespace, svcPorts, serviceEntry.Location, resolution,
-		exportTo, labelSelectors, serviceEntry.SubjectAltNames, creationTime)...)
+		exportTo, labelSelectors, serviceEntry.SubjectAltNames, creationTime, cfg.Labels)...)
 	return out
 }
 
 func buildServices(hostAddresses []*HostAddress, namespace string, ports model.PortList, location networking.ServiceEntry_Location,
 	resolution model.Resolution, exportTo map[visibility.Instance]bool, selectors map[string]string, saccounts []string,
-	ctime time.Time) []*model.Service {
+	ctime time.Time, labels map[string]string) []*model.Service {
```

Comment:
Let's pass the annotations too... I'm not sure this case is label or annotation, but probably we should consistently preserve both.